src/theory/datatypes/datatypes_rewriter.h

```diff
@@ -84,6 +84,13 @@ class DatatypesRewriter : public TheoryRewriter
    * reason about the original selector term e.g. via congruence.
    */
   static Node expandApplySelector(Node n, bool sharedSel);
+  /**
+   * Expand updater term. Given n = (APPLY_UPDATER{C} t s), this method returns
+   *   (ITE (APPLY_TESTER{C} t)
+   *      (C (APPLY_SELECTOR{1} t)... s ... (APPLY_SELECTOR{n} t))
+   *       t).
```

Comment:
shouldn't the index of the term to be replaced be there somewhere? Also, do not use `n` for the last selector.

Reply:
The updater is indexed by a selector, I'll update the doc.